includes/Modules/Idea_Hub.php

```diff
@@ -146,12 +156,102 @@ function( $maybe_empty, $postarr ) {
 	 */
 	public function register() {
 		$this->register_scopes_hook();
+		if ( $this->is_connected() ) {
+			/**
+			 * Show admin notices on the posts page if we have saved / new ideas.
+			 */
+			add_filter( 'googlesitekit_admin_notices', $this->get_method_proxy( 'admin_notice_idea_hub_ideas' ) );
+		}
 
 		$this->post_name_setting->register();
 		$this->post_text_setting->register();
 		$this->post_topic_setting->register();
 	}
 
+	/**
+	 * Shows admin notification for idea hub ideas on post list screen.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $notices Array of admin notices.
+	 * @return array Array of admin notices.
+	 */
+	private function admin_notice_idea_hub_ideas( $notices ) {
+		global $post_type;
+		$current_screen = get_current_screen();
+		if ( is_null( $current_screen ) || 'edit-post' !== $current_screen->id || 'post' !== $post_type ) {
+			return $notices;
+		}
+		$dismissed_items_instance = new Dismissed_Items( $this->user_options );
+		$transients               = new Transients( $this->context );
+
+		$notice_settings = array(
+			'saved' => array(
+				'text' => esc_html__( 'Need some inspiration? Revisit your saved ideas in Site Kit', 'google-site-kit' ),
+				'cta'  => esc_html__( 'See saved ideas', 'google-site-kit' ),
+				'link' => admin_url( 'admin.php?page=googlesitekit-dashboard#new-ideas' ),
+			),
+			'new'   => array(
+				'text' => esc_html__( 'Need some inspiration? Here are some new ideas from Site Kit’s Idea Hub', 'google-site-kit' ),
+				'cta'  => esc_html__( 'See new ideas', 'google-site-kit' ),
+				'link' => admin_url( 'admin.php?page=googlesitekit-dashboard#saved-ideas' ),
+			),
+		);
+
+		$dismissed_items = $dismissed_items_instance->get_dismissed_items();
+		$ideas           = $transients->get( self::TRANSIENT_IDEAS );
+
+		if ( false === $ideas ) {
+			$ideas = array(
+				'saved-ideas' => $this->get_data( 'saved-ideas' ),
+				'new-ideas'   => $this->get_data( 'new-ideas' ),
+			);
+
+			$transients->set( self::TRANSIENT_IDEAS, $ideas, DAY_IN_SECONDS );
+		}
+
+		$has_saved_ideas = count( $ideas['saved-ideas'] ) > 0;
+		$has_new_ideas   = count( $ideas['new-ideas'] ) > 0;
+
+		if ( ! $has_saved_ideas && in_array( 'saved-ideas', $dismissed_items, true ) ) {
+			// Saved items no longer need to be dismissed as there are none currently.
+			$this->dismissed_items->delete( 'saved-ideas' );
+		}
```

Comment:
> ... there is one particular circumstance when it should be cleared: If the user makes a change that results in them to no longer have any saved ideas (for example unsaving the only saved idea or creating a draft post from the only saved idea), the dismissal should be cleared. In other words, once the user later saves something again, the notice should re-appear.

We should clear the `saved-ideas` dismissal in the appropriate endpoint handler when a user saves or removes an idea from the saved ideas list.

Reply:
I think this might have to be done later, and I can remove this portion entirely:
saving / unsaving an idea would be via `POST:update-idea-state` which is currently pending.

Creating a draft post from the only saved idea, I think this may need a call to `POST:create-idea-draft-post` but that's insufficient to remove the idea from the saved ideas.

Thoughts here?

Reply:
Ok, let's use your approach for now. I think your version is the best option for now.